src/core/sock.h

```diff
@@ -78,6 +79,9 @@ struct nn_sock
     /*  Next endpoint ID to assign to a new endpoint. */
     int eid;
 
+    /*  Count of active holds against the socket. */
+    int holds;
```

Comment:
How about making this an atomic counter, using atomic increment/decrement? Is it possible that this alone could elimiate other mutexing? (probably not, but I have not thought through the problem; you're in the zone and could probably give a quick answer)

Could we gain anything else from using atomic counter here, further than a pedantic semantic? Conversely, do there exist new potential bugs from not using atomic counter?